### DISCUSSION TITLE
Which is a good and compatibile web UI for Quartz?

### DISCUSSION THREAD

**Discussion by biapar** [quartznet/quartznet#1079] [Q&A] [answered]
Which is a good and compatibile web UI for Quartz?

**Comment by lahma** [comment, marked as answer]
There are currently three viable options which you can evaluate:

* [Crystal Quartz Panel](https://github.com/guryanovev/CrystalQuartz)
* [Quartzmin](https://github.com/jlucansky/Quartzmin)
* [QuartzNetWebConsole](https://github.com/mausch/QuartzNetWebConsole) (hasn't been updated for a while, might not work with 3.x at all)

Some day there might be one build-in too.

**Reply by falvarez1** [reply]
Quartzmin hasn't been worked on in a while, a fork was created and is up to date with .Net 5 here https://github.com/maikebing/SilkierQuartz

**Reply by GoOnYahGoodThing** [reply]
Hi, 
Is this discussion closed now?
I just had a look at the GUI options above and they seem to be a bit old now.
Just wondering is there a newer GUI project that anyone knows about?
Maybe something based on .net core 6 ?

Thanks

**Reply by lahma** [reply]
If by old you mean mature and proven, yes I guess so. Is there something you are missing feature-wise?

**Comment by GoOnYahGoodThing** [comment]
Oops, if I insulted somebody by asking that question then I apologise!
That was not my intention.
I looked at the github for Quartzmin and it appeared to be untouched for 3 years so I figured it was abandoned like a lot of open source projects. SilkierQuartz also didn't appear to have a lot of recent changes too.
But if it there are still people around it then great.

**Reply by lahma** [reply]
Yes I was a bit snarky, generally projects have less updates when they feel reaching feature complete 😉

**Reply by GoOnYahGoodThing** [reply]
Just for curiosity.. how is Quartzmin (and silkier) implemented? I see files that are .hbs files etc?
I'm just getting to grips with .net core so I'm learning about razor pages etc.. but haven't seen info about .hbs etc?

Thanks

**Reply by lahma** [reply]
I believe those are handlebars files, the repo has this in readme:
> SilkierQuartz was created with Semantic UI and Handlebars.Net as the template engine.

**Reply by GoOnYahGoodThing** [reply]
Right.. I see that now.. ok.. thanks.

Can you tell me is there any info on how to add jobs etc the way they are done in the SilkierQuartz.Example/startup.cs file?
Seems to be a different way of doing it from whats in the Quartz tutorial.
I've been trying just to add a new job type and then having it available in the GUI so I could create a trigger for it.. but it looks like I have to create a job and a trigger and start it all off in one go.. that wouldn't suit the kind of processing jobs I would be hoping to create...

**Reply by lahma** [reply]
I'm afraid I cannot give support on 3rd party UI. But generally you can always register a job into database and mark it as durable. This way it won't be removed after there are no triggers referencing it. Then after you have the job(s) you need, you can add triggers referencing it, maybe with trigger specific job data or then use job specific shared data properties via job data map.

**Reply by GoOnYahGoodThing** [reply]
Thanks Marko,
I'm sorry, I shouldn't be asking you about this 3rd Party thing.. 
I will try to see if there is anyone on the Silkier repo who might answer.
Thanks

**Reply by GoOnYahGoodThing** [reply]
Sorry if I am pushing my luck here.. but maybe I can just write the code for a less complicated GUI myself.
In that case.. can you actually register the job as durable without kicking it off ? Or does it need to start once in the scheduler?
Like, does a job always need a trigger to get registered?
And is there some way to get the list of registered jobs out of the scheduler ? ( so I could show a list on a page for example)

Thanks again... I'll go away after this.. I promise...

**Reply by lahma** [reply]
Yes durable job is something that stays - even without triggers referencing it. You need to delete it manually when no longer needed. Trigger on the other hand always needs a target job. IScheduler interface has ways to retrieve both jobs and triggers.
